src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -258,19 +262,30 @@ bool emberAfOtaSoftwareUpdateRequestorClusterAnnounceOtaProviderCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::OtaSoftwareUpdateRequestor::Commands::AnnounceOtaProvider::DecodableType & commandData)
 {
-    EmberAfStatus status;
-    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    auto & metadataForNode = commandData.metadataForNode;
 
-    if (requestor != nullptr)
+    if (commandObj == nullptr)
     {
-        status = requestor->HandleAnnounceOTAProvider(commandObj, commandPath, commandData);
+        ChipLogError(SoftwareUpdate, "Invalid commandObj, cannot process AnnounceOtaProvider");
+        return false;
     }
-    else
+
+    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
     {
-        status = EMBER_ZCL_STATUS_FAILURE;
+        commandObj->AddStatus(commandPath, Status::UnsupportedCommand);
+        return true;
     }
 
-    emberAfSendImmediateDefaultResponse(status);
+    if (metadataForNode.HasValue() && metadataForNode.Value().size() > kMaxMetadataLen)
+    {
+        ChipLogError(Zcl, "Metadata size %zu exceeds max %zu", metadataForNode.Value().size(), kMaxMetadataLen);
```

Comment:
We shouldn't use "%zu" on servers: it does not exist in some small libc implementations.  Probably better to just use %u and cast to unsigned.